### DISCUSSION TITLE
Creating build ( android .aab, ios xCode archive )

### DISCUSSION THREAD

**Discussion by Kasik-D** [microsoft/rnx-kit#2548] [Q&A]
Hello, I recently started using rnx-kit, and I really don't understand more things. 
I hope you can help me with this. 

I have the react-native version 0.69.5, I did everything as the documentation said. But I don't undestand rnx-kit used to create .aab file or ios archive. 

For build android .aab I just use `"build": "cd android && ./gradlew bundleRelease && cd ../"`, 
For ios I use Xcode

My package.json configuration 

```
 "rnx-kit": {
    "bundle": {
      "treeShake": true,
      "plugins": [
        ["@rnx-kit/metro-plugin-duplicates-checker", { "throwOnError": false }],
        ["@rnx-kit/metro-plugin-typescript", { "throwOnError": true }],
        [
          "@rnx-kit/metro-plugin-cyclic-dependencies-detector",
          { "throwOnError": false }
        ]
      ],
      "entryFile": "index.js",
      "targets": ["android", "ios"],
      "platforms": {
        "android": {
          "bundleOutput": "dist/main.android.bundle",
          "sourcemapOutput": "dist/main.android.bundle.map",
          "assetsDest": "dist/res"
        },
        "ios": {
          "bundleOutput": "dist/main.ios.jsbundle",
          "sourcemapOutput": "dist/main.ios.jsbundle.map",
          "assetsDest": "dist"
        }
      }
    },
    "server": {
      "plugins": [
        ["@rnx-kit/metro-plugin-cyclic-dependencies-detector", { "throwOnError": false } ],
        [
          "@rnx-kit/metro-plugin-duplicates-checker",
          {
            "ignoredModules": ["react-is"],
            "throwOnError": false
          }
        ],
        "@rnx-kit/metro-plugin-typescript"
      ]
    }
  }
```

I want the rnx-kit to be used when creating builds

**Comment by tido64** [comment]
Hi @Kasik-D, `rnx-kit` currently does not replace the need for native tools like those for building `.aab` and `.ipa` files. Please refer to the official documentations for how to get those built, for instance:
- [Building your first app bundle](https://medium.com/androiddevelopers/building-your-first-app-bundle-bbcd228bf631) (Android)
- [Prepare you app and share it with your team, beta testers, and customers](https://developer.apple.com/documentation/xcode/distribution) (iOS)

**Reply by Kasik-D** [reply]
@tido64 Thanks for the answer, but it's very confusing, why do we need to create these files if we can't use them for builds
![image](https://github.com/microsoft/rnx-kit/assets/61847567/dced7eb7-3bdd-40d5-aaee-d2130215d223)

**Reply by tido64** [reply]
Those are JavaScript bundles. They are basically all the JavaScript code you wrote and dependencies, merged into a single file. You ship your JavaScript bundle with your app as assets. Your app initializes the React Native runtime that will load the bundle. The bundles are not directly related to `.aab` or .`ipa`.
